user_guide_src/source/helpers/url_helper.rst

```diff
@@ -0,0 +1,325 @@
+##########
+URL Helper
+##########
+
+The URL Helper file contains functions that assist in working with URLs.
+
+.. contents::
+  :local:
+
+.. raw:: html
+
+  <div class="custom-index container"></div>
+
+Loading this Helper
+===================
+
+This helper is loaded using the following code::
+
+	helper('url');
+
+Available Functions
+===================
+
+The following functions are available:
+
+.. php:function:: site_url([$uri = ''[, $protocol = NULL]])
+
+	:param	string	$uri: URI string
+	:param	string	$protocol: Protocol, e.g. 'http' or 'https'
+	:returns:	Site URL
+	:rtype:	string
+
+	Returns your site URL, as specified in your config file. The index.php
+	file (or whatever you have set as your site **index_page** in your config
+	file) will be added to the URL, as will any URI segments you pass to the
+	function, plus the **url_suffix** as set in your config file.
+
+	You are encouraged to use this function any time you need to generate a
+	local URL so that your pages become more portable in the event your URL
+	changes.
+
+	Segments can be optionally passed to the function as a string or an
+	array. Here is a string example::
+
+		echo site_url('news/local/123');
+
+	The above example would return something like:
+	*http://example.com/index.php/news/local/123*
+
+	Here is an example of segments passed as an array::
+
+		$segments = array('news', 'local', '123');
```

Comment:
Hmm. I had forgotten site_url could take an array. It won't currently, I don't believe...